include/fastpath/fastpath.h

```diff
@@ -46,9 +46,11 @@ static inline void cancelIPC_fp(tcb_t *dest)
         endpoint_ptr_set_state(ep_ptr, EPState_Idle);
     }
 
-    reply_t *reply = REPLY_PTR(thread_state_get_replyObject(dest->tcbState));
-    if (reply != NULL) {
-        reply_unlink(reply, dest);
+    if (thread_state_get_tsType(dest->tcbState) == ThreadState_BlockedOnReceive) {
```

Comment:
I think that this if could be an assert. cancelIPC_fp is only called when the dest tcb is  bound to the notification and is blocked on an ep (and not blocked on a notification), so the only possible state would be  ThreadState_BlockedOnReceive. If this code was enabled for x86 virtualization configurations then maybe it could be inside a guest, but there are other parts of the signal fastpath that would break in that case.

Reply:
I wasn't sure enough to turn this into an assert, good point. Will have another closer look.

Reply:
You are right, this can be an assert. In fact the only path leading to this function being called explicitly checks for `BlockedOnReceive` and returns early if that is not the case:

https://github.com/seL4/seL4/blob/c679fe77d6441cd412dd253d953ee833dd25d740/src/fastpath/fastpath.c#L618-L623

Reply:
(added assert and rebase)